build/cmake/programs/CMakeLists.txt

```
@@ -38,6 +38,8 @@ if (UNIX)
     add_custom_target(unzstd ALL ${CMAKE_COMMAND} -E create_symlink zstd unzstd DEPENDS zstd COMMENT "Creating unzstd symlink")
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/zstdcat DESTINATION "bin")
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/unzstd DESTINATION "bin")
+    install(FILES ${CMAKE_SOURCE_DIR}/../../programs/zstdgrep DESTINATION "bin" PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
+    install(FILES ${CMAKE_SOURCE_DIR}/../../programs/zstdless DESTINATION "bin" PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
```

Perhaps there's too much toggling or permissions?
```suggestion
    install(FILES ${CMAKE_SOURCE_DIR}/../../programs/zstdless DESTINATION "bin")
```

---

well, the default permissions is OWNER_WRITE, OWNER_READ, GROUP_READ, and WORLD_READ
[cmake install](https://cmake.org/cmake/help/v3.13/command/install.html)
which means the users can not directly use zstdless and zstdgrep after make install.
Users need to chmod for zstdless and zstdgrep.
I tested on Ubuntu, Redhat, Centos and suse, and all of them show the same result if I didn't add the execute permissions
`-bash: /usr/local/bin/zstdless: Permission denied`

I think this should be done in cmake scripts.

And of course, the commits should be squashed. Thank you so much

---

Can you explain why these permissions need to be explicitly set here, while they don't on the `zstd` install command? I guess it's because `zstd` is a `TARGETS` object, while these are `FILES` objects. Would it make more sense for these to be `TARGETS` themselves?

Or maybe these should be `PROGRAMS`?

> The PROGRAMS form is identical to the FILES form except that the default permissions for the installed file also include OWNER_EXECUTE, GROUP_EXECUTE, and WORLD_EXECUTE. This form is intended to install programs that are not targets, such as shell scripts. Use the TARGETS form to install targets built within the project.

---

@felixhandte I checked the camke documentation and ask some guys familiar with CMake. I think you are right. These should be `PROGRAMS`. Thank you so much.